src/Compilers/Core/Portable/SourceGeneration/Nodes/SourceOutputNode.cs

```diff
@@ -21,6 +21,8 @@ internal sealed class SourceOutputNode<TInput> : IIncrementalGeneratorOutputNode
 
         private readonly IncrementalGeneratorOutputKind _outputKind;
 
+        private readonly string _name;
```

Comment:
>_name [](http://example.com/codeflow?start=32&length=5)

It looks like `_name` is a constant and only used in `UpdateStateTable`. Consider inlining in that method and removing the field.